### PR TITLE
Change edit links and de-clutter summary page

### PR DESCRIPTION
**Pull request by Amirroracle**
-Moved some of the css to the styles.js, should perhaps start thinking of indexing/de-cluttering the styles.js page since it is beginning to get a tad large.
-Set up a better abstraction for the table in the summary page. A lot less clutter.
-Added the links according to the design doc for the 'Edit' and 'Change' links in the summary page. 

Todo:
-wait until @pcraig3's merge for the calendar to go through and add the calendar data to the availability row. 
-add tests for the new components I added.

### CODE REVIEW

File: web/src/Summary.js [hunk 1]
```diff
@@ -73,97 +39,27 @@ export const Summary = ({
...
-        )}
-      </Column2>
-      <Column3>
-        <TextLink className={change} href="#">
-          <Trans>Change</Trans>
-        </TextLink>
-      </Column3>
-    </Row>
+    <UserInfoRow
+      header="Full name"
```
Comment (pcraig3) [inline web/src/Summary.js]: I think it's better if these are `<Trans>` elements, so we can change the name between English and French. 

ie
```
header={<Trans>Full name</Trans>} 
```

And then you could update the proptypes at the bottom to be 
`header: PropTypes.element.isRequired`.

File: web/src/Summary.js [hunk 2]
```diff
@@ -73,97 +39,27 @@ export const Summary = ({
...
+      secondColumn={fullName}
+      thirdColumn="Edit"
```
Comment (pcraig3) [inline web/src/Summary.js]: In the new mockup, the text for the third column is always the same. 
The link changes, but the link text is the same.

image

File: web/src/TableContents.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      </H2>
+    </Column1>
+    <Column2>{secondColumn}</Column2>
+    <Column3>
+      <TextLink className={change} href="/calendar">
+        <Trans>{thirdColumn}</Trans>
+      </TextLink>
+    </Column3>
+  </Row>
+)
```
Comment (pcraig3) [inline web/src/TableContents.js]: There's a lot of similar code for `Column1`, `Column2` (now that the dates are in) and `Column3` across both of these components. You could probably generalise this further. 
(but up to you; I'm not so bothered)

File: web/src/TableContents.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        <Trans>{header}</Trans>
+      </H2>
+    </Column1>
+    <Column2>
+      <p>{secondColumn}</p>
+    </Column2>
+    <Column3>
+      <TextLink href="/register">
+        <Trans>{thirdColumn}</Trans>
+      </TextLink>
```
Comment (pcraig3) [inline web/src/TableContents.js]: We want `NavLinks` here.

If you can delete `TextLink` outright, I would be pleased. But if it's still being used somewhere else, then nvm.

File: web/src/TableContents.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  header: PropTypes.string,
+  secondColumn: PropTypes.object,
+  thirdColumn: PropTypes.string,
+}
+
+UserInfoRow.propTypes = {
+  header: PropTypes.string,
+  secondColumn: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  thirdColumn: PropTypes.string,
+}
```
Comment (pcraig3) [inline web/src/TableContents.js]: Both DateRow and UserInfoRow can accept the same proptypes now.

I think they both take: 
```
{
  header: PropTypes.element.isRequired,
  secondColumn: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,   
}
```

And then for the third column, you could pass in a string if you wanted (which would be the href).
ie 
```
  thirdColumn: PropTypes.string.isRequired
```

File: web/src/Summary.js (hunk 1 above)
Comment (Amirroracle) [reply web/src/Summary.js]: Nvm, I think I found a decent abstraction

File: web/src/Summary.js (hunk 2 above)
Comment (Amirroracle) [reply web/src/Summary.js]: yup, good point

File: web/src/Summary.js [hunk 6]
```diff
@@ -73,97 +40,51 @@ export const Summary = ({
...
+        </NavLink>
+      }
+    />
+    <SummaryRow
+      header={<Trans>Paper file number</Trans>}
+      secondColumn={paperFileNumber}
+      thirdColumn={
+        <NavLink to="/register">
+          <Trans>Change</Trans>
+        </NavLink>
```
Comment (pcraig3) [inline web/src/Summary.js]: Either this, or your `thirdColumn` could just pass in a string that's the `to` string.

So you could be like 

```
// Summary.js
<SummaryRow
  ...
  thirdColumn="/register"
/>

// TableContents.js
const SummaryRow ...
  <Row>
    <Column3>
      <NavLink to={thirdColumn}>
        <Trans>Change</Trans>
      </NavLink>
    </Column3>
  </Row>
```

Up to you.

File: web/src/TableContents.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  </Row>
+)
+
+SummaryRow.propTypes = {
+  header: PropTypes.object.isRequired,
+  secondColumn: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  thirdColumn: PropTypes.object.isRequired,
+}
+
+export { SummaryRow }
```
Comment (pcraig3) [inline web/src/TableContents.js]: This file is pretty small now and only contains 1 component. 

I would suggest:
1. put this code straight into `Summary.js` and delete this file.
2. rename the file to `SummaryRow.js`

**Comment by pcraig3** [comment]
rebased and ready for liftoff! 🚀

**Pull request merged**